wallet/wallet.go

```diff
@@ -107,6 +107,17 @@ func (w *HotWallet) NextAddress() types.Address {
 	return addr
 }
 
+// Adresses returns all addresses generated by the wallet.
+func (w *HotWallet) Addresses() []types.Address {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	a := make([]types.Address, w.store.SeedIndex())
+	for i := uint64(0); i < w.store.SeedIndex(); i++ {
+		a[i] = types.StandardAddress(w.seed.PublicKey(i))
+	}
+	return a
+}
```

Comment:
This might be premature optimization, but technically only the `SeedIndex` needs to be covered by the mutex (and deriving addresses isn't the fastest operation, so we should avoid doing it under lock if possible).

```go
w.mu.Lock()
addrs := make([]types.Address, w.store.SeedIndex())
w.mu.Unlock()
for i := range addrs {
    addrs[i] = types.StandardAddress(w.seed.PublicKey(uint64(i)))
}
return addrs
```